app/src/main/java/com/portfolio/tasky/usecases/NetworkStatus.kt

```diff
@@ -0,0 +1,6 @@
+package com.portfolio.tasky.usecases
+
+sealed class NetworkStatus{
+    object Available : NetworkStatus()
+    object Unavailable : NetworkStatus()
+}
```

Comment:
If you only need these 2 statuses, you can also just use a boolean `isAvailable` for the sake of simplicity. Also, this could be an `enum class` since your different statuses don't need different constructors (which would be a reason to use a sealed class)

Reply:
I would request to keep this implementation as it is for now. I am gradually learning best practices,  and I really find this implementation easy to understand. I also assure that I will change the implementation to a better one in a future PR